Automattic/wp-calypso
# Social: Improve the way Jetpack Social products are displayed.

The Jetpack Social products are shown using its own component, which we wrote just to render social products. After receiving feedback from multiple people, we have decided to display the social products in the same way as we display the backup products. 



## Proposed Changes

* Show the Jetpack Social plans using radio buttons
* If your existing social subscription is a monthly one, currently on prod, we just take you to the calypso page after clicking on "Manage Subscription". In this PR, we open the modal and display the monthly variants of the prices. The behaviour is unchanged if your existing social subscription is a yearly one. 
* I have also added the changes to include the introductory cost calculation for the monthly plan

## Testing Instructions



Testing the pricing page for siteless checkouts
- Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the Jetpack Social Basic and Advanced plans. 
<img width="1142" alt="Screenshot 2023-02-08 at 18 35 25" src="https://user-images.githubusercontent.com/6594561/217538352-be7b90ae-43bd-40b0-a174-5e3db6757bb6.png">
-  Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the VaultPress backup plan. This step is to make sure that the  VaultPress backup plan, the only other product with an introductory offer gets checked out correctly. 
- Go to http://jetpack.cloud.localhost:3000/pricing and make sure you're able to checkout the Jetpack Boost plan. This step is to make sure that the non-introductory offers plans are being checked out correctly. 


Testing the pricing page for sites with existing social plans, please boot 4 new sites as described below and don't use an existing site since free trials won't be available to the site that has already used them. 

- Boot up a new jetpack site and buy the Jetpack Social Basic monthly plan.
- Boot up a new jetpack site and buy the Jetpack Social Basic yearly plan.
- Boot up a new jetpack site and buy the Jetpack Social Advanced monthly plan.
- Boot up a new jetpack site and buy the Jetpack Social Advanced yearly plan.
- For each of the above sites, Go to http://jetpack.cloud.localhost:3000/pricing/your-jeptpack.site.domain?site=your-jeptpack.site.domain and click on the manage subscription button next to "Social". 
<img width="688" alt="Screenshot 2023-02-08 at 18 43 53" src="https://user-images.githubusercontent.com/6594561/217540075-a075cadb-a763-47e8-a8d1-44b08962a175.png">

- For the product you already own, you will be able to see the "Manage Subscription" button and for the product you don't own, you will see the add to cart button. 
<img width="561" alt="Screenshot 2023-02-08 at 18 47 04" src="https://user-images.githubusercontent.com/6594561/217540944-b1da00e5-76ea-4d99-93ff-16576b6d940f.png">



## Pre-merge Checklist
